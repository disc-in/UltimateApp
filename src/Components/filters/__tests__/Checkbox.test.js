import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import Checkbox from '../Checkbox';

describe('<Checkbox />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<Checkbox title="OK" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when active', () => {
    const { toJSON } = render(<Checkbox title="OK" active />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls delegates onPress prop', async () => {
    const onPress = jest.fn();

    const { getByText } = render(<Checkbox title="Easy" onPress={onPress} />);

    await fireEvent.press(getByText('Easy'));

    expect(onPress).toHaveBeenCalled();
  });
});
