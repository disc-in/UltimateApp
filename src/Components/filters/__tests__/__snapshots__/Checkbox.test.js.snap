// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders correctly 1`] = `
<View
  style={
    Object {
      "flexBasis": "50%",
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#ededed",
        "borderRadius": 3,
        "borderWidth": 0,
        "margin": 5,
        "marginLeft": 10,
        "marginRight": 10,
        "opacity": 1,
        "padding": 0,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "#43484d",
            "fontWeight": "normal",
            "marginLeft": 10,
            "marginRight": 10,
            "textTransform": "capitalize",
          }
        }
        testID="checkboxTitle"
      >
        OK
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Checkbox /> renders correctly when active 1`] = `
<View
  style={
    Object {
      "flexBasis": "50%",
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": true,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#ededed",
        "borderRadius": 3,
        "borderWidth": 0,
        "margin": 5,
        "marginLeft": 10,
        "marginRight": 10,
        "opacity": 1,
        "padding": 0,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#2a9e91",
              "fontSize": 24,
            },
            Object {
              "minWidth": 24,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ÔÅÜ
      </Text>
      <Text
        style={
          Object {
            "color": "#43484d",
            "fontWeight": "normal",
            "marginLeft": 10,
            "marginRight": 10,
            "textTransform": "capitalize",
          }
        }
        testID="checkboxTitle"
      >
        OK
      </Text>
    </View>
  </View>
</View>
`;
